misk-core/src/main/kotlin/misk/sampling/RateLimiter.kt

```diff
@@ -107,6 +108,33 @@ class RateLimiter private constructor(
     }
   }
 
+  /**
+   * Returns the permits remaining, given a time unit and timeout, after tryAcquire() has been invoked.
```

Comment:
I think this comment is a bit misleading. There's no requirement for this method to be called before or after `tryAcquire` has been invoked; it can be called any time, even if `tryAcquire` is never invoked.

Perhaps something like this?

> Returns the maximum number of permits that could have been acquired by a call to [tryAcquire], assuming the caller passed the same [timeout] and [unit].